Sp 1065

tenthe opened the pull request:
<!--
  ~ Licensed to the Apache Software Foundation (ASF) under one or more
  ~ contributor license agreements.  See the NOTICE file distributed with
  ~ this work for additional information regarding copyright ownership.
  ~ The ASF licenses this file to You under the Apache License, Version 2.0
  ~ (the "License"); you may not use this file except in compliance with
  ~ the License.  You may obtain a copy of the License at
  ~
  ~    http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~ Unless required by applicable law or agreed to in writing, software
  ~ distributed under the License is distributed on an "AS IS" BASIS,
  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
  ~
  -->
  
  <!--
Thanks for contributing! Here are some tips you can follow to help us incorporate your contribution quickly and easily:
1. If this is your first time, please read our contributor guidelines:
    - https://streampipes.apache.org/getinvolved.html
    - https://cwiki.apache.org/confluence/display/STREAMPIPES/Getting+Started
2. Make sure the PR title is formatted like: `[#<GitHub issue id>] PR title ...`
3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., `[WIP][#<GitHub issue id>] PR title ...`.
4. Please write your PR title to summarize what this PR proposes/fixes.
5. Link the PR to the corresponding GitHub issue (if present) in the `Development` section in the right menu bar. 
6. Be sure to keep the PR description updated to reflect all changes.
7. If possible, provide a concise example to reproduce the issue for a faster review.
8. Make sure tests pass via `mvn clean install`.
9. (Optional) If the contribution is large, please file an Apache ICLA
    - http://apache.org/licenses/icla.pdf
-->

### Purpose
<!--
Please clarify what changes you are proposing and describe how those changes will address the issue.
Furthermore, describe potential consequences the changes might have.
-->
Fixes #1065 by adding e2e tests for the different formats that can be used in the adapters.
Also enabled the live preview feature for all formats.

### Remarks
<!--
Is there anything left we need to pay attention on?
Are there some references that might be important? E.g. links to Confluence, or discussions
on the mailing list or GitHub.
-->
This PR also removes the underscore dependency and replaces it with jackson-dataformat-xml, which was already registered in the pom.
I also had to adjust some of the parser implementations. This PR provides the basis for further refactoring of those parsers.
PR introduces (a) breaking change(s): no

PR introduces (a) deprecation(s): no

dominikriemer commented on streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/format/geojson/GeoJsonParser.java:
```diff
@@ -85,66 +85,85 @@ public void parse(InputStream data, EmitBinaryEvent emitBinaryEvent) throws Pars
       LineString lineString = (LineString) geoFeature.getGeometry();
       eventProperties.add(JsonEventProperty.getEventProperty("coorindatesLineString", lineString.getCoordinates()));
-
+      sampleValues.put("coorindatesLineString", new GuessTypeInfo("java.lang.Double", lineString.getCoordinates()));
```
is this a type and should be named `coordinates`?

tenthe replied:
Thanks, you are right. It seems to be a typo.

dominikriemer commented on streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/format/geojson/GeoJsonParser.java:
```diff
@@ -85,66 +85,85 @@ public void parse(InputStream data, EmitBinaryEvent emitBinaryEvent) throws Pars
       MultiPoint multiPoint = (MultiPoint) geoFeature.getGeometry();
       eventProperties.add(JsonEventProperty.getEventProperty("coorindatesMultiPoint", multiPoint.getCoordinates()));
-
+      sampleValues.put("coorindatesMultiPoint", new GuessTypeInfo("java.lang.Double", multiPoint.getCoordinates()));
```
same for all types

dominikriemer commented on streampipes-extensions/streampipes-connect-adapters-iiot/pom.xml:
```diff
@@ -127,6 +127,12 @@
+                <exclusion>
+                    <groupId>com.fasterxml.woodstox</groupId>
+                    <artifactId>woodstox-core</artifactId>
+                </exclusion>
```
Does this exclusion have any impact on the S7 driver?

tenthe replied:
Hopefully not, we have to validate it with a PLC as soon as it is merged. This is included due to some convergence errors

tenthe commented:
@dominikriemer I have addressed your comments. I also extracted some common strings into a new class `GeoJsonConstants`.
We should improve the implementation further, but this should reduce the risk of typos for now.

dominikriemer reviewed (approved):
Thanks!
